frontend/public/components/pod.tsx

```diff
@@ -104,15 +109,40 @@ const tableColumnClasses = [
   classNames('pf-m-hidden', { 'pf-m-visible-on-xl pf-u-w-10-on-2xl': showMetrics }),
   classNames('pf-m-hidden', { 'pf-m-visible-on-xl pf-u-w-10-on-2xl': showMetrics }),
   classNames('pf-m-hidden', 'pf-m-visible-on-2xl pf-u-w-10-on-2xl'),
+  classNames('pf-m-hidden', 'pf-m-visible-on-xl'),
+  classNames('pf-m-hidden', 'pf-m-visible-on-xl'),
+  classNames('pf-m-hidden', 'pf-m-visible-on-xl'),
+
   Kebab.columnClass,
 ];
 
 const kind = 'Pod';
 
 const podRowStateToProps = ({ UI }) => ({
   metrics: UI.getIn(['metrics', 'pod']),
+  columnFilters: UI.getIn(['columnManagement']),
 });
 
+type PodTableDataProps = {
+  className: string;
+  columnName: string;
+  columnFilter: any;
+};
+
+const PodTableData: React.FC<PodTableDataProps> = ({
+  className,
+  columnName,
+  columnFilter,
+  children,
+}) => {
+  return (
+    ((!_.isEmpty(columnFilter)
+      ? columnFilter?.find((col) => col.title === columnName && col.visible === true)
+      : true) && <TableData className={className}>{children}</TableData>) ||
```

Comment:
```suggestion
    const visible = _.isEmpty(columnFilter) || columnFilter?.find((col) => col.title === columnName && col.visible === true);
    return visible ? <TableData className={className}>{children}</TableData> : null;
```

Reply:
We should generalize this so we don't need this logic in every list view.

Reply:
ok, removed need for PodTableData and added all necessary logic into TableData to generalize it.